plugins/field-date/src/field_date.ts

```diff
@@ -0,0 +1,263 @@
+ *
+ * @param num A one or two digit number.
+ * @returns The two digit string for a number.
+ */
+function getTwoDigits(num: number): string {
+  const numStr = `${num}`;
+  return numStr.length === 1 ? `0${numStr}` : numStr;
+}
+
+/**
+ * Get the string formatted locally to the user.
+ * @param dateString A string in the format 'yyyy-mm-dd'
+ * @returns the locale date string for the date.
+ */
+function getLocaleDateString(dateString: string): string {
+  // NOTE: `date.toLocaleDateString()` will be the day before for western dates
+  // due to an unspecified time & timezone assuming midnight at GMT+0.
+  const date = new Date(dateString);
+
+  // NOTE: This format varies per region.
+  // Ex: "5/12/2023", "12/05/2023", "12.5.2023", "2023/5/12", "१२/५/२०२३"
+  const language = navigator.language ?? 'en-US';
+  // See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat#using_options
+  return Intl.DateTimeFormat(language, {
+    // NOTE: Print the date for GMT+0 since the date object assumes midnight at 
+    // GMT+0.
+    timeZone: 'UTC',
+  }).format(date);
+}
+
+/**
+ * NOTE: There are a few minor ways to tweak the datepicker CSS, though they're 
+ * not consistent across browsers.
+ * @see{@link https://developer.mozilla.org/en-US/docs/Learn/Forms/Property_compatibility_table_for_form_controls#date_pickers}
+ * 
+ * Below are a few ways this can be tweaked on *some* browsers:
+ * Blockly.Css.register(`
+ * ::-webkit-datetime-edit { }
+ * ::-webkit-datetime-edit-fields-wrapper { }
+ * ::-webkit-datetime-edit-text { }
+ * ::-webkit-datetime-edit-month-field { }
+ * ::-webkit-datetime-edit-day-field { }
+ * ::-webkit-datetime-edit-year-field { }
+ * ::-webkit-inner-spin-button { }
+ * ::-webkit-calendar-picker-indicator { }
+ * `);
+ */
+if (Blockly.utils.userAgent.MAC) {
+  // NOTE: By default, 4 px padding total are added within the User Agent 
+  // Shadow Content on Safari on MAC. Remove the padding so the inner input 
+  // matches the outer input's height and, by extension, the height of the text 
+  // node.
+  Blockly.Css.register(`
+input.blocklyDateInput::-webkit-datetime-edit,
+input.blocklyDateInput::-webkit-datetime-edit-month-field,
+input.blocklyDateInput::-webkit-datetime-edit-day-field,
+input.blocklyDateInput::-webkit-datetime-edit-year-field {
+  padding: 0;
+}
+`);
+}
+
+Blockly.fieldRegistry.register('field_date', FieldDate);
+
+/**
+ * A config object for defining a field date.
+ */
+export type FieldDateConfig = Blockly.FieldTextInputConfig;
+
+/**
+ * Options used to define a field date from JSON.
+ */
+export interface FieldDateFromJsonConfig extends FieldDateConfig {
+  date?: string;
+  // NOTE: spellcheck is defined for FieldInput though irrelevant for FieldDate.
+  // TODO: Confirm 'never' options. Should these be allowed with FieldDate? If
+  // so, how would they be used?
+  spellcheck?: never;
+  tooltip?: never;
```

Comment:
Having spellcheck as `never` makes sense, but I don't understand `never`ing tooltip.

Reply:
Looks like the previous implementation never passed the config, with tooltip, along. I updated it to do so now, though, and confirmed in manual tests the tooltip appears over the click target.